src/backend/pipeline/sw_vacuum.c

```diff
@@ -234,6 +234,13 @@ ShouldVacuumSWTuple(SWVacuumContext *context, HeapTupleData *tuple)
 	if (!context)
 		return false;
 
+	/*
+	 * The combiner could have updated this tuple before actually committing/aborting,
+	 * so we ignore it in that case.
+	 */
+	if (tuple->t_data->t_infomask & (HEAP_XMAX_COMMITTED | HEAP_XMAX_INVALID))
```

Comment:
Should this be `HEAP_XMIN_COMMITTED | HEAP_XMIN_INVALID`? 

`xmax` is only set when the tuple is deleted, or has been replaced by an updated tuple. `xmin` is the xid of the transaction that created the tuple.

http://www.postgresql.org/docs/9.1/static/ddl-system-columns.html